python/lsst/faro/measurement/TractTableValueMeasurement.py

```diff
@@ -0,0 +1,159 @@
+            " to determine standard colors",
+        default=('u', 'g', 'r', 'i', 'z', 'y'),
+    )
+    format_column = pexConfig.Field(
+        dtype=str,
+        doc="Format of the full column names including the band",
+        default="{band}_{column}",
+    )
+    prefixes_column = pexConfig.ListField(
+        dtype=str,
+        doc="Column name prefixes to ignore when applying special formatting rules",
+        default=[f'{x.value.label}_' for x in SourceType],
+    )
+    row = pexConfig.Field(
+        dtype=int,
+        doc="Index of the row to retrieve the value from",
+        optional=False,
+    )
+    unit = pexConfig.Field(
+        dtype=str,
+        doc="The astropy unit of the metric value",
+        default='',
+    )
+
+    def _format_column(self, band: str, column: str):
+        prefix = ''
+        for prefix_column in self.prefixes_column:
+            if column.startswith(prefix_column):
+                prefix = prefix_column
+                column = column[len(prefix):]
+                break
+        if column.startswith('color_'):
+            column = f'color_{self.band_order[self.band_order.index(band)-  1]}_m_{band}_{band}_{column[6:]}'
+        if column.startswith('flux_'):
+            column = f'flux_{band}_{column[5:]}'
+        elif column.startswith('mag_'):
+            column = f'mag_{band}_{column[4:]}'
+        return self.format_column.format(band=band, column=f'{prefix}{column}')
+
+
+class TractTableValueMeasurementTask(MetricTask):
+    """Measure a metric from a single row and combination of columns in a table."""
+
+    ConfigClass = TractTableValueMeasurementConfig
+    _DefaultName = "TractTableValueMeasurementTask"
+
+    def run(self, table, bands, name_metric):
+        unit = u.Unit(self.config.unit)
+        measurements = [None]*len(bands)
+        columns = list(self.config.action.columns)
+        for idx, band in enumerate(bands):
+            row = table.iloc[[self.config.row]].rename(
+                columns={self.config._format_column(band, column): column
+                         for column in columns}
+            )
+            value = self.config.action(row).iloc[0]
+            measurements[idx] = Measurement(name_metric, value*unit)
+        return Struct(measurement=measurements)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        try:
+            inputs = butlerQC.get(inputRefs)
+            bands = [x.dataId['band'] for x in [y for y in outputRefs][0][1]]
+            columns_base = list(self.config.action.columns)
+            columns_in = []
+            for band in bands:
+                columns_in.extend(self.config._format_column(band, column)
+                                  for column in columns_base)
+
+            # TODO: does it make sense to validate this now or just leave it to the run call?
+            # assert all(column in inputs['columns'] for column in columns_in)
```

Comment:
Perhaps this check is more appropriate here? It seems more like a validation of the formatting of the inputs to the run method rather than something fundamental to the calculation itself.

Reply:
Ah sorry, I should clarify - if any of `columns_in` are not in the table, then the call to `inputs['table'].get(parameters={'columns': columns_in})` three lines later will fail. So I figured there's not really any point in duplicating a check that the get method does.